server/server.js

```diff
@@ -218,6 +219,30 @@ Server.prototype._handleResourceMessage = function(socket, message, messageType)
   }
 };
 
+// Process the resource messages associated with a single client
+Server.prototype._handleResourceMessages = function (socket, client) {
+  var subscriptions = client.subscriptions;
+  var presences = client.presences;
+
+  // Pause events on the inbound socket
+  Pauseable.pause(socket);
+
+  for (var key in subscriptions) {
+    var message = subscriptions[key];
+    var messageType = this._getMessageType(message.to);
```

Comment:
This might be expensive, as Vanchi already mentioned before. We should probably create a cache for types by resource (on a different PR).